src/pages/SearchPage/SearchPage.js

```diff
@@ -1,38 +1,96 @@
-import React, { useState } from 'react';
-import { useHistory } from 'react-router-dom';
+import React, { useState, useEffect } from 'react';
+import { useHistory, useParams } from 'react-router-dom';
 import SearchPageWrapper from './SearchPage.style';
 import SearchBar from '../SearchBar/SearchBar';
 import Spinner from '../../components/Spinner/Spinner';
 
+
+const ONE_YEAR_IN_SECONDS = 60 * 60 * 24 * 365;
+
+async function fetchData(url) {
+  const response = await fetch(url);
+  return response.json();
+}
+
 function SearchPage() {
   const history = useHistory();
-
   const [isLoading, setIsLoading] = useState(false);
+  const [searchResults, setSearchResults] = useState([]);
+  const path = useParams().subreddit;
+  const [redditAPIObj, setRedditAPIObj] = useState({
+    type: 'submission',
+    params: {
+      subreddit: 'javascript',
+      sort: 'desc',
+      sort_type: 'created_utc',
+      // Current time in seconds minus one year
+      after: Math.round((new Date()).getTime() / 1000) - ONE_YEAR_IN_SECONDS,
+      size: 500,
+    },
+  });
 
-  async function fetchData() {
-    setIsLoading(true);
-    const response = await fetch(
-      'https://www.reddit.com/r/javascript/search.json?q=oop&limit=5',
-      { mode: 'cors' },
-    );
-    if (response.status === 200) {
+  function buildRedditAPILink(APIObj) {
+    // Build an API link based on the params given. Will look similar to:
+    // https://api.pushshift.io/reddit/search/submission/?subreddit=javascript$sort=desc$sort_type=created_utc$after=1554462807$size=500
+    let APILink = `https://api.pushshift.io/reddit/search/${APIObj.type}/`;
+    Object.keys(APIObj.params).forEach((key, index) => {
+      let urlOperator = '';
+      if (index === 0) {
+        urlOperator = '?';
+      } else {
+        urlOperator = '&';
+      }
+      APILink += `${urlOperator}${key}=${APIObj.params[key]}`;
+    });
+    return APILink;
+  }
+
+  async function handleSearch() {
```

Comment:
I moved the `handleSearch` function out of the component. This way it's a constant and doesn't need to be added to the dependencies of `useEffect`. To make it a pure function though you need to move all external dependencies like `setIsLoading` and `setSearchResults` to the `useEffect`.